docs/content/latest/explore/ysql-language-features/triggers.md

```diff
@@ -0,0 +1,239 @@
+
+When YugabyteDB is being used by different applications, triggers allow you to keep the cross-functional behavior within the database that is performed automatically every time the table data is modified. In addition, triggers let you maintain data integrity rules that can only be implemented at the database level. 
+
+When using triggers, it is important to be aware of their existence and understand their logic. Otherwise, it is difficult to predict the timing and effects of data changes.
+
+## Creating Triggers
+
+Creating a trigger in YSQL is a two-step process: you start by creating a trigger function via the `CREATE FUNCTION` statement, and then you bind the trigger function to a table using the `CREATE TRIGGER` statement.
+
+The `CREATE FUNCTION` statement has the following syntax:
+
+```sql
+CREATE FUNCTION trigger_function() 
+RETURNS TRIGGER LANGUAGE PLPGSQL
+AS $$
+BEGIN
+   -- ...
+END;
+$$
+```
+
+*trigger_function* obtains information about the environment that is invoking it via a container populated with local variables.
+
+The `CREATE TRIGGER` statement has the following syntax:
+
+```sql
+CREATE TRIGGER tr_name 
+{BEFORE | AFTER} { event } 
+ON tbl_name [FOR [EACH] { ROW | STATEMENT }]
+       EXECUTE PROCEDURE trigger_function
+```
+
+The trigger *tr_name* fires before or after *event* which can be set to `INSERT` , `DELETE`, `UPDATE`, or `TRUNCATE`. *tbl_name* represents the table associated with the trigger. If you use the `FOR EACH ROW` clause, the scope of the trigger would be one row. If you use the `FOR EACH STATEMENT` clause, the trigger would be fired for each transaction. *trigger_function* represents the procedure to be performed when the trigger is fired.
+
+Suppose you work with a database that includes the following table populated with data:
+
+```sql
+CREATE TABLE employees (
+  employee_no integer PRIMARY KEY,
+  name text,
+  department text
+);
+
+INSERT INTO employees (employee_no, name, department) 
+  VALUES 	
+  (1221, 'John Smith', 'Marketing'),
+  (1222, 'Bette Davis', 'Sales'),
+  (1223, 'Lucille Ball', 'Operations'),
+  (1224, 'John Zimmerman', 'Sales'); 
+```
+
+If an employee is transferred to a different department, the change is recorded in a table called `employee_dept_changes`, as shown in the following example:
+
+```sql
+CREATE TABLE employee_dept_changes (
+  employee_no integer NOT NULL,
+  name text,
+  department text,
```

Comment:
nit: will leave this up to you but maybe this example would be more illustrative if this has two columns
`old_department` and `new_department` and then the trigger updates both to capture the actual change.

Reply:
I'll keep it as is for now (mainly because I need to move to the platform docs ASAP) and then come back at a later time to change this example.